deoxys/src/modes.rs

```diff
@@ -294,38 +294,49 @@ where
         subkeys: &Array<DeoxysKey, B::SubkeysSize>,
         tag: &mut Tag,
     ) {
-        if !buffer.is_empty() {
-            tweak[0] = TWEAK_M;
+        if buffer.is_empty() {
+            return;
+        }
+        tweak[0] = TWEAK_M;
 
-            for (index, data) in buffer.chunks(16).enumerate() {
-                // Copy block number
-                tweak[8..].copy_from_slice(&(index as u64).to_be_bytes());
+        let mut chunks = buffer.chunks_exact(16);
```

Comment:
You could use [`Array::slice_as_chunks`](https://docs.rs/hybrid-array/latest/hybrid_array/struct.Array.html#method.slice_as_chunks) here